### PR TITLE
feat(platform): FormGroup layout support

### PR DESCRIPTION
**Pull request by fkolar**
Initial implementation for the FormGroup Layout based on the [Spec](https://github.com/SAP/fundamental-ngx/wiki/Platform:-FormGroup-Layout-Technical-Design)

Also disabling one lint rule due the problem which is still present in the tslint [64](https://github.com/mgechev/codelyzer/issues/64) issue

Once it is merged we will do next updated  based on the latest FormGroup/FormField/FormControl adjustments we have made with @KevinOkamoto.

We expect that added `package.json` dependencies for the flexgrid will be removed and replaced with fundamental-styles layout implementations once it is available. 

Current implemented as based on `core` form's directives for which I have filed also several issues:
https://github.com/SAP/fundamental-ngx/issues/2134
https://github.com/SAP/fundamental-ngx/issues/2133

Also we need to move all the inputs under this `form/` dir

@DeepakSap14  let's merge this first as I have made proper structure for the form that you can follow, so you can put your code in here. 

#### Please provide a link to the associated issue.
#2131

### CODE REVIEW

File: libs/platform/src/lib/components/form/base.input.ts [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+    id: string = this.defaultId;
+
+    @Input()
+    get name(): string {
+        return this._name ? this._name : this.id;
+    }
+
+    set name(value: string) {
+        this._name = value;
+        this.stateChanges.next('rb set name');
```
Comment (KevinOkamoto) [inline libs/platform/src/lib/components/form/base.input.ts]: Is the `name` property something we need to track changes on? Will the `name` of an input field change in an application or is there an operation that we need to do on setting of the input `name`?

File: libs/platform/src/lib/components/form/input/input.component.html [hunk 2]
```diff
@@ -0,0 +1,16 @@
+<!--
+  Does it make sense to have some other state than error?
+-->
+<input #elemRef fd-form-control
+       [(ngModel)]="value"
+       [ngModelOptions]="{standalone: true}"
+       [attr.id]="id"
+       [attr.type]="type"
```
Comment (KevinOkamoto) [inline libs/platform/src/lib/components/form/input/input.component.html]: Need to add `[attr.name]` property.

File: libs/platform/src/lib/components/form/base.input.ts (hunk 1 above)
Comment (fkolar) [reply libs/platform/src/lib/components/form/base.input.ts]: I have this only for debugging. But there are properties that are expected to change after is initialized and properties that never change. Next push is is going to reflect this.

**Comment by fkolar** [comment]
Screen Shot 2020-03-09 at 10 15 25 PM
Screen Shot 2020-03-09 at 10 15 16 PM
Screen Shot 2020-03-09 at 10 15 08 PM



```html
<fdp-form-group #fg [multiLayout]="true"  >
    <fdp-form-field #ffl1 [id]="'serialNum'" label="Serial Number" hint="Hint for SN " zone="zLeft"
                    rank="10" required="true">
        <fdp-input [formControl]="ffl1.formControl"></fdp-input>
    </fdp-form-field>

    <fdp-form-field #ffl2 [id]="'employeeId'" label="Employee ID"  hint="Hint for one " zone="zRight"
                    rank="10">
        <fdp-input [formControl]="ffl2.formControl"></fdp-input>
    </fdp-form-field>


    <ng-template #i18n let-errros>
        <span *ngIf="errros.required">
            Value is required
        </span>
        <span *ngIf="errros.maxlength">
            You reached maxium allowed length. Required is {{errros.maxlength.requiredLength}}
            current is
            {{errros.maxlength.actualLength}}
        </span>
    </ng-template>
</fdp-form-group>

```

**Comment by fkolar** [comment]
For the `hintPlacement` we will come back to this once it starts to work in the core.. See the ticket I opened

**Comment by fkolar** [comment]
@JKMarkowski and @droshev adding to other approvers to pass PR check.

**Pull request merged**